### PR TITLE
Update from microsoft.sql 2021-05-01-preview to base branch

### PR DESCRIPTION
**Pull request by ericshape**
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

1. add a new API version 2021-05-01-preview for SQL RP.
2. Carry all the swagger updates from SQL RP for the 2021-05-01-preview release

compare with 2021-02-01-preview, there are three new groups of APIs added:

- preview/2021-05-01-preview/DistributedAvailabilityGroups.json
- preview/2021-05-01-preview/ServerConnectionPolicies.json
- preview/2021-05-01-preview/ServerTrustCertificates.json

### Changelog
Add a changelog entry for this PR by answering the following questions:
  1. What's the purpose of the update?
      - [ ] new service onboarding
      - [x] new API version
      - [ ] update existing version for new feature
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify
  2. When are you targeting to deploy the new service/feature to public regions? Please provide the date or, if the date is not yet available, the month.
  3. When do you expect to publish the swagger? Please provide date or, the the date is not yet available, the month.
  4. If updating an existing version, please select the specific langauge SDKs and CLIs that must be refreshed after the swagger is published.
      - [ ] SDK of .NET (need service team to ensure code readiness)
      - [ ] SDK of Python
      - [ ] SDK of Java
      - [ ] SDK of Js
      - [ ] SDK of Go
      - [ ] PowerShell
      - [ ] CLI
      - [ ] Terraform
      - [x] No refresh required for updates in this PR

### Contribution checklist:
- [x] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of "no breaking changes"
- [x] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [x] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist 

> **Applicability**: :warning: 
>
> If your changes encompass only the following scenarios, you should SKIP this section, as these scenarios do not require ARM review.
> - Change to data plane APIs 
> - Adding new properties 
> - All removals

Otherwise your PR may be subject to ARM review requirements. Complete the following:
- [x] Check this box if any of the following apply to the PR so that label “WaitForARMFeedback” will be added automatically to begin ARM API Review. Failure to comply may result in delays to the manifest.
  - Adding a new service
  - Adding new API(s)
  - Adding a new API version
    -[ ] To review changes efficiently, ensure you copy the existing version into the new directory structure for first commit and then push new changes, including version updates, in separate commits.

- [x] Ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If any of the following scenarios apply to the PR, request approval from the Breaking Change Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 
- [ ] Removing API(s) in a stable version
- [ ] Removing properties in a stable version
- [ ] Removing API version(s) in a stable version
- [ ] Updating API in a stable or public preview version with Breaking Change Validation errors
- [ ] Updating API(s) in public preview over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Addition details on the process and office hours are on the [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

**Comment by ericshape** [comment]
The validation pipeline has been passed. 
The failed ones are not due to this new API version change.

@zhenglaizhang  can you help to take a look? Thanks!

**Comment by ericshape** [comment]
This PR also carries some differences between the latest branch and Azure:dev-sql-Microsoft.Sql-2021-05-01-preview-release

**Comment by zhenglaizhang** [comment]
Hi @jianyexi do you know how to fix the track 2 validation failure https://github.com/Azure/azure-rest-api-specs/pull/16184/checks?check_run_id=3724619480 ?

**Comment by zhenglaizhang** [comment]
@fengzhou-msft could you help check .net sdk build failure https://github.com/Azure/azure-rest-api-specs/pull/16184/checks?check_run_id=3724632772 ?

**Comment by ericshape** [comment]
@jianyexi @fengzhou-msft 
can you please take a look? Thanks!
we expect this PR can be merged soon and the 2021-05-01-preview will be released by September 30th.

**Comment by jianyexi** [comment]
> Hi @jianyexi Jianye Xi FTE do you know how to fix the track 2 validation failure https://github.com/Azure/azure-rest-api-specs/pull/16184/checks?check_run_id=3724619480 ?

Can be ignored

**Comment by ericshape** [comment]
compare with 2021-02-01-preview, there are three new groups of APIs added:

> preview/2021-05-01-preview/DistributedAvailabilityGroups.json
> preview/2021-05-01-preview/ServerConnectionPolicies.json
> preview/2021-05-01-preview/ServerTrustCertificates.json

File: specification/sql/resource-manager/Microsoft.Sql/preview/2021-05-01-preview/DatabaseExtensions.json [hunk 1]
```diff
@@ -277,6 +354,45 @@
           "description": "Error message.",
           "type": "string",
           "readOnly": true
+        },
+        "queuedTime": {
+          "description": "Queued time.",
+          "type": "string",
```
Comment (TimLovellSmith) [inline specification/sql/resource-manager/Microsoft.Sql/preview/2021-05-01-preview/DatabaseExtensions.json:360]: is it e.g. "format": "date-time"?

https://swagger.io/docs/specification/data-models/data-types/#:~:text=OpenAPI%20defines%20the%20following%20built-in%20string%20formats%3A%20date,by%20RFC%203339%2C%20section%205.6%2C%20for%20example%2C%202017-07-21T17%3A32%3A28Z

Comment (ericshape) [reply specification/sql/resource-manager/Microsoft.Sql/preview/2021-05-01-preview/DatabaseExtensions.json:360]: this is intended for sql usage.

File: specification/sql/resource-manager/Microsoft.Sql/preview/2021-05-01-preview/Databases.json [hunk 2]
```diff
@@ -787,6 +787,44 @@
         }
       }
     },
+    "DatabaseIdentity": {
```
Comment (TimLovellSmith) [inline specification/sql/resource-manager/Microsoft.Sql/preview/2021-05-01-preview/Databases.json:790]: Could you leverage https://github.com/Azure/azure-rest-api-specs/blob/main/specification/common-types/resource-management/v3/managedidentity.json instead?
It greatly simplifies reviewing for consistency.

File: specification/sql/resource-manager/Microsoft.Sql/preview/2021-05-01-preview/ManagedInstanceOperations.json [hunk 3]
```diff
@@ -326,9 +326,20 @@
...
+        "stepStartTime": {
+          "format": "date-time",
+          "type": "string"
+        },
+        "stepEndTime": {
+          "format": "date-time",
+          "type": "string"
+        },
+        "timeElapsed": {
+          "type": "string"
```
Comment (TimLovellSmith) [inline specification/sql/resource-manager/Microsoft.Sql/preview/2021-05-01-preview/ManagedInstanceOperations.json:341]: any particular format? Or description of one?

File: specification/sql/resource-manager/Microsoft.Sql/preview/2021-05-01-preview/Databases.json (hunk 2 above)
Comment (ericshape) [reply specification/sql/resource-manager/Microsoft.Sql/preview/2021-05-01-preview/Databases.json:790]: we might not carry https://github.com/Azure/azure-rest-api-specs/blob/main/specification/common-types/resource-management/v3/managedidentity.json this time, due to it is just added 16 days ago. 
however, all our swagger files in this PR is auto-generated by SQL repo. Change the reference would need to go through all the test process again. 
We will address this issue in the separate PR.

File: specification/sql/resource-manager/Microsoft.Sql/preview/2021-05-01-preview/ManagedInstances.json [hunk 4]
```diff
@@ -752,16 +752,32 @@
...
-            "ZRS"
+            "Geo",
+            "Local",
+            "Zone",
+            "GeoZone"
+          ],
+          "type": "string",
+          "readOnly": true,
+          "x-ms-enum": {
+            "name": "CurrentBackupStorageRedundancy",
```
Comment (TimLovellSmith) [inline specification/sql/resource-manager/Microsoft.Sql/preview/2021-05-01-preview/ManagedInstances.json:766]: Could there just be one Definition of BackupStorageRedundancy enum type, which is referenced for both currentBackupStorageRedundancy and requestedBackupStorageRedundancy properties? So that the values are obviously interchangeable? Or are they likely to diverge in the future?

File: specification/sql/resource-manager/Microsoft.Sql/preview/2021-05-01-preview/ManagedInstances.json [hunk 5]
```diff
@@ -877,6 +897,39 @@
         }
       }
     },
+    "ServicePrincipal": {
```
Comment (TimLovellSmith) [inline specification/sql/resource-manager/Microsoft.Sql/preview/2021-05-01-preview/ManagedInstances.json:900]: This looks a lot like SystemAssignedServiceIdentity in
https://github.com/Azure/azure-rest-api-specs/blob/main/specification/common-types/resource-management/v3/managedidentity.json
except this one adds clientId. Why is there a divergence?

File: specification/sql/resource-manager/Microsoft.Sql/preview/2021-05-01-preview/ServerConnectionPolicies.json [hunk 6]
```diff
@@ -0,0 +1,269 @@
...
+    "ServerConnectionPolicy": {
+      "description": "A server connection policy",
+      "type": "object",
+      "allOf": [
+        {
+          "$ref": "../../../common/v1/types.json#/definitions/ProxyResource"
+        }
+      ],
+      "properties": {
+        "location": {
```
Comment (TimLovellSmith) [inline specification/sql/resource-manager/Microsoft.Sql/preview/2021-05-01-preview/ServerConnectionPolicies.json:211]: Usually nested proxy resources shouldn't need a location property.

File: specification/sql/resource-manager/Microsoft.Sql/preview/2021-05-01-preview/ServerConnectionPolicies.json [hunk 7]
```diff
@@ -0,0 +1,269 @@
...
+    }
+  },
+  "parameters": {
+    "ServerNameParameter": {
+      "name": "serverName",
+      "in": "path",
+      "description": "The name of the server.",
+      "required": true,
+      "type": "string",
+      "x-ms-parameter-location": "method"
```
Comment (TimLovellSmith) [inline specification/sql/resource-manager/Microsoft.Sql/preview/2021-05-01-preview/ServerConnectionPolicies.json:255]: I'm confused about why "x-ms-parameter-location": "method"? Isn't it coming from the URI, as is usual, not the method?

File: specification/sql/resource-manager/Microsoft.Sql/preview/2021-05-01-preview/ServerConnectionPolicies.json (hunk 6 above)
Comment (ericshape) [reply specification/sql/resource-manager/Microsoft.Sql/preview/2021-05-01-preview/ServerConnectionPolicies.json:211]: it is intended to prevent breaking change from 2014-04-01.

File: specification/sql/resource-manager/Microsoft.Sql/preview/2021-05-01-preview/TimeZones.json [hunk 8]
```diff
@@ -141,7 +141,7 @@
       }
     },
     "TimeZone": {
-      "description": "Time Zone.",
+      "description": "TimeZone.",
```
Comment (TimLovellSmith) [inline specification/sql/resource-manager/Microsoft.Sql/preview/2021-05-01-preview/TimeZones.json:144]: Fuller description?

File: specification/sql/resource-manager/Microsoft.Sql/preview/2021-05-01-preview/DatabaseExtensions.json (hunk 1 above)
Comment (theFakeFiliron) [reply specification/sql/resource-manager/Microsoft.Sql/preview/2021-05-01-preview/DatabaseExtensions.json:360]: The format for the date would be like: "queuedTime": "2/2/2020 8:33:27 PM"
Does this answer your question?

Comment (theFakeFiliron) [reply specification/sql/resource-manager/Microsoft.Sql/preview/2021-05-01-preview/DatabaseExtensions.json:360]: Logic is same as the one for [ImportDatabase.json](https://msdata.visualstudio.com/Database%20Systems/_git/DsMainDev?path=%2FSql%2Fxdb%2Fmanifest%2Fsvc%2Fmgmt%2FWebApiHosting%2FARM%2FAutoGenSwagger%2FMicrosoft.Sql%2Fpreview%2F2020-02-02-preview%2Fexamples%2FImportDatabase.json&_a=contents&version=GBmaster)

File: specification/sql/resource-manager/Microsoft.Sql/preview/2021-05-01-preview/TimeZones.json (hunk 8 above)
Comment (ericshape) [reply specification/sql/resource-manager/Microsoft.Sql/preview/2021-05-01-preview/TimeZones.json:144]: I will fix it

File: specification/sql/resource-manager/Microsoft.Sql/preview/2021-05-01-preview/ServerConnectionPolicies.json [hunk 9]
```diff
@@ -0,0 +1,269 @@
...
+        "operationId": "ServerConnectionPolicies_Get",
+        "parameters": [
+          {
+            "$ref": "../../../common/v1/types.json#/parameters/ResourceGroupParameter"
+          },
+          {
+            "$ref": "#/parameters/ServerNameParameter"
+          },
+          {
+            "name": "connectionPolicyName",
```
Comment (TimLovellSmith) [inline specification/sql/resource-manager/Microsoft.Sql/preview/2021-05-01-preview/ServerConnectionPolicies.json:34]: Could reduce duplication by promoting this into a Definition and referencing it.

File: specification/sql/resource-manager/Microsoft.Sql/preview/2021-05-01-preview/ServerConnectionPolicies.json (hunk 6 above)
Comment (TimLovellSmith) [reply specification/sql/resource-manager/Microsoft.Sql/preview/2021-05-01-preview/ServerConnectionPolicies.json:211]: How so, I thought this is a new resource type you're adding in this version?

File: specification/sql/resource-manager/Microsoft.Sql/preview/2021-05-01-preview/ServerConnectionPolicies.json [hunk 10]
```diff
@@ -0,0 +1,269 @@
...
+          "description": "The server connection type.",
+          "enum": [
+            "Default",
+            "Redirect",
+            "Proxy"
+          ],
+          "type": "string",
+          "x-ms-enum": {
+            "name": "ServerConnectionType",
+            "modelAsString": false
```
Comment (TimLovellSmith) [inline specification/sql/resource-manager/Microsoft.Sql/preview/2021-05-01-preview/ServerConnectionPolicies.json:197]: modelAsString true is usually recommended for enums that might grow in future

File: specification/sql/resource-manager/Microsoft.Sql/preview/2021-05-01-preview/ServerTrustCertificates.json [hunk 11]
```diff
@@ -0,0 +1,297 @@
...
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Sql/managedInstances/{managedInstanceName}/serverTrustCertificates": {
+      "get": {
+        "tags": [
+          "ServerTrustCertificates"
+        ],
+        "description": "Gets a list of server trust certificates that were uploaded from box to the given Sql Managed Instance.",
```
Comment (TimLovellSmith) [inline specification/sql/resource-manager/Microsoft.Sql/preview/2021-05-01-preview/ServerTrustCertificates.json:181]: 'from box' seems a bit irrelevant, you could drop it. The important point to clarify would be what these certificates mean to the server.

File: specification/sql/resource-manager/Microsoft.Sql/preview/2021-05-01-preview/ServerTrustCertificates.json [hunk 12]
```diff
@@ -0,0 +1,297 @@
...
+      "properties": {
+        "properties": {
+          "$ref": "#/definitions/ServerTrustCertificateProperties",
+          "description": "Resource properties.",
+          "x-ms-client-flatten": true
+        }
+      }
+    },
+    "ServerTrustCertificatesListResult": {
+      "description": "A list of server trust certificates in instance.",
```
Comment (TimLovellSmith) [inline specification/sql/resource-manager/Microsoft.Sql/preview/2021-05-01-preview/ServerTrustCertificates.json:257]: What does 'server trust certificate' mean?

File: specification/sql/resource-manager/Microsoft.Sql/preview/2021-05-01-preview/DatabaseExtensions.json (hunk 1 above)
Comment (TimLovellSmith) [reply specification/sql/resource-manager/Microsoft.Sql/preview/2021-05-01-preview/DatabaseExtensions.json:360]: I'm suggesting you can either add a 'format' property in the swagger here to give some hinting to swagger consumers, code generators etc, as to the actual format, or at least mention it in the description, so people don't have to guess.

File: specification/sql/resource-manager/Microsoft.Sql/preview/2021-05-01-preview/ServerConnectionPolicies.json (hunk 6 above)
Comment (tunafish88) [reply specification/sql/resource-manager/Microsoft.Sql/preview/2021-05-01-preview/ServerConnectionPolicies.json:211]: Hi Tim, connection policies is a resource that we already support in the 2014 version. However we had to introduce it here as part of this new API to allow for async behavior because we had some ICM in the past which led us to deciding this API should be Async. With this change the idea is customers will have the option to consume the new ask to call this api without having to chance any of their code, and so we've tried to keep it matching with the 2014 version.

File: specification/sql/resource-manager/Microsoft.Sql/preview/2021-05-01-preview/DatabaseExtensions.json (hunk 1 above)
Comment (ericshape) [reply specification/sql/resource-manager/Microsoft.Sql/preview/2021-05-01-preview/DatabaseExtensions.json:360]: fix in DSMain

File: specification/sql/resource-manager/Microsoft.Sql/preview/2021-05-01-preview/TimeZones.json (hunk 8 above)
Comment (ericshape) [reply specification/sql/resource-manager/Microsoft.Sql/preview/2021-05-01-preview/TimeZones.json:144]: fix in DSMain

File: specification/sql/resource-manager/Microsoft.Sql/preview/2021-05-01-preview/ServerConnectionPolicies.json (hunk 10 above)
Comment (ericshape) [reply specification/sql/resource-manager/Microsoft.Sql/preview/2021-05-01-preview/ServerConnectionPolicies.json:197]: fix in DSMain

File: specification/sql/resource-manager/Microsoft.Sql/preview/2021-05-01-preview/DatabaseExtensions.json (hunk 1 above)
Comment (ericshape) [reply specification/sql/resource-manager/Microsoft.Sql/preview/2021-05-01-preview/DatabaseExtensions.json:360]: we might fix it in 2021-08-01

File: specification/sql/resource-manager/Microsoft.Sql/preview/2021-05-01-preview/ManagedInstanceOperations.json (hunk 3 above)
Comment (krivi37) [reply specification/sql/resource-manager/Microsoft.Sql/preview/2021-05-01-preview/ManagedInstanceOperations.json:341]: timeElapsed is a string that returns the number of minutes passed from stepStartTime

File: specification/sql/resource-manager/Microsoft.Sql/preview/2021-05-01-preview/ManagedInstances.json (hunk 4 above)
Comment (svmicano) [reply specification/sql/resource-manager/Microsoft.Sql/preview/2021-05-01-preview/ManagedInstances.json:766]: We have to have both currentBackupStorageRedundancy and requestedBackupStorageRedundancy because we are soon going to enable backup residency change feature. That way  currentBackupStorageRedundancy and requestedBackupStorageRedundancy will not point to the same value while the change is in progress. For Managed Instance, this is a size of data operation and can last up to 24h. Also the parameters are aligned with sql database parameters now and that was a decision from all of our PMs

Comment (TimLovellSmith) [reply specification/sql/resource-manager/Microsoft.Sql/preview/2021-05-01-preview/ManagedInstances.json:766]: @svmicano Let me try to clarify. I'm not saying that they point to the same value. I'm saying that it looks like they point to the same *type* of value, so you shouldn't really need to declare two separate enum types.

File: specification/sql/resource-manager/Microsoft.Sql/preview/2021-05-01-preview/ManagedInstanceOperations.json (hunk 3 above)
Comment (TimLovellSmith) [reply specification/sql/resource-manager/Microsoft.Sql/preview/2021-05-01-preview/ManagedInstanceOperations.json:341]: OK, could you add that as the description?

File: specification/sql/resource-manager/Microsoft.Sql/preview/2021-05-01-preview/ServerConnectionPolicies.json (hunk 6 above)
Comment (TimLovellSmith) [reply specification/sql/resource-manager/Microsoft.Sql/preview/2021-05-01-preview/ServerConnectionPolicies.json:211]: @tunafish88 Thanks, makes sense.

File: specification/sql/resource-manager/Microsoft.Sql/preview/2021-05-01-preview/ServerTrustCertificates.json (hunk 12 above)
Comment (nemijato) [reply specification/sql/resource-manager/Microsoft.Sql/preview/2021-05-01-preview/ServerTrustCertificates.json:257]: I would change this to:
"A list of the server trust certificates which are used for secure communication between SQL On-Prem instance and the given Sql Managed Instance"

File: specification/sql/resource-manager/Microsoft.Sql/preview/2021-05-01-preview/ServerTrustCertificates.json (hunk 11 above)
Comment (nemijato) [reply specification/sql/resource-manager/Microsoft.Sql/preview/2021-05-01-preview/ServerTrustCertificates.json:181]: I would change this to:
"Gets a list of the server trust certificates which are used for secure communication between SQL On-Prem instance and the given Sql Managed Instance"

File: specification/sql/resource-manager/Microsoft.Sql/preview/2021-05-01-preview/DatabaseExtensions.json (hunk 1 above)
Comment (theFakeFiliron) [reply specification/sql/resource-manager/Microsoft.Sql/preview/2021-05-01-preview/DatabaseExtensions.json:360]: Yes, we can fix it in 2021-08-01 if that's ok with you.

File: specification/sql/resource-manager/Microsoft.Sql/preview/2021-05-01-preview/ServerConnectionPolicies.json (hunk 10 above)
Comment (ericshape) [reply specification/sql/resource-manager/Microsoft.Sql/preview/2021-05-01-preview/ServerConnectionPolicies.json]: updated to true

File: specification/sql/resource-manager/Microsoft.Sql/preview/2021-05-01-preview/TimeZones.json (hunk 8 above)
Comment (ericshape) [reply specification/sql/resource-manager/Microsoft.Sql/preview/2021-05-01-preview/TimeZones.json]: updated description.

File: specification/sql/resource-manager/Microsoft.Sql/preview/2021-05-01-preview/ManagedInstances.json (hunk 4 above)
Comment (svmicano) [reply specification/sql/resource-manager/Microsoft.Sql/preview/2021-05-01-preview/ManagedInstances.json:766]: They do point to the same type of value but I was not aware that there is a better way to do it. I aligned the api with SQLDB design which is like this. Since these APIs are chasing Ignite, is it OK to create a follow up PR for this when we get time?

File: specification/sql/resource-manager/Microsoft.Sql/preview/2021-05-01-preview/ManagedInstances.json (hunk 5 above)
Comment (ericshape) [reply specification/sql/resource-manager/Microsoft.Sql/preview/2021-05-01-preview/ManagedInstances.json:900]: sorry, the owner is not located. 
can we fix this issue in the later PR?

File: specification/sql/resource-manager/Microsoft.Sql/preview/2021-05-01-preview/ServerConnectionPolicies.json (hunk 9 above)
Comment (ericshape) [reply specification/sql/resource-manager/Microsoft.Sql/preview/2021-05-01-preview/ServerConnectionPolicies.json:34]: we will fix it in the upcoming PR. 
all the swagger files are auto-generated. we have to fix it in SQL repo as upstream. 
given that this issue does not impact the functionality and breaking change, can we fix this issue in the next PR?

File: specification/sql/resource-manager/Microsoft.Sql/preview/2021-05-01-preview/ServerTrustCertificates.json (hunk 11 above)
Comment (ericshape) [reply specification/sql/resource-manager/Microsoft.Sql/preview/2021-05-01-preview/ServerTrustCertificates.json:181]: I will directly add the description in the swagger PR to pass the review.
@nemijato will send out a PR to fix it in SQL repo as upstream?

File: specification/sql/resource-manager/Microsoft.Sql/preview/2021-05-01-preview/ServerTrustCertificates.json (hunk 12 above)
Comment (ericshape) [reply specification/sql/resource-manager/Microsoft.Sql/preview/2021-05-01-preview/ServerTrustCertificates.json:257]: I will directly update the description in the swagger PR to pass the review.
@nemijato will send out a PR to fix it in SQL repo as upstream?

File: specification/sql/resource-manager/Microsoft.Sql/preview/2021-05-01-preview/ManagedInstances.json (hunk 5 above)
Comment (TimLovellSmith) [reply specification/sql/resource-manager/Microsoft.Sql/preview/2021-05-01-preview/ManagedInstances.json:900]: Getting all services using the same definitions for system assigned identity helps ensure API consistency across Azure. And comparing the types manually makes for a much harder review process and raises the risk of breaking changes because inconsistences exist. For example you don't use the 'uuid' format specifier whereas common types. (Is that going to be considered breaking? I'm not even sure.) Is that what you prefer?

If you can get the definition right this first time, then future breaking changes checks could at least be helping you ensure you are keeping it consistent, even though your auto-generated SDK isn't generating it the same yet, or even though you go back to autogenerating it without common types, you should get property level consistency checking.

File: specification/sql/resource-manager/Microsoft.Sql/preview/2021-05-01-preview/ManagedInstances.json (hunk 4 above)
Comment (TimLovellSmith) [reply specification/sql/resource-manager/Microsoft.Sql/preview/2021-05-01-preview/ManagedInstances.json:766]: @svmicano  In that case it could become a small, but SDK-breaking change to unify the types in your next PR, so I'd recommend you instead fix it in this PR. By creating a Definition to $ref.

File: specification/sql/resource-manager/Microsoft.Sql/preview/2021-05-01-preview/ManagedInstances.json (hunk 5 above)
Comment (vmanukyan1) [reply specification/sql/resource-manager/Microsoft.Sql/preview/2021-05-01-preview/ManagedInstances.json:900]: This ServicePrincipal object is introduced in scope of windows authentication feature. It is the representation of AAD application, which will be used for Kerberos authentication.

Regarding to 'uuid' types: Here we are using strings, as for managing AAD application we are using MSAL library which returns/expects strings for these parameters. We stored these parameters as string, to get rid of uuid <-> string conversions.

Comment (TimLovellSmith) [reply specification/sql/resource-manager/Microsoft.Sql/preview/2021-05-01-preview/ManagedInstances.json:900]: @vmanukyan1 That's actually a rather good practical criticism of using UUID format specifier. I'm OK with not-fixing this for now based on that.

/cc @pilor

File: specification/sql/resource-manager/Microsoft.Sql/preview/2021-05-01-preview/ManagedInstances.json (hunk 4 above)
Comment (ericshape) [reply specification/sql/resource-manager/Microsoft.Sql/preview/2021-05-01-preview/ManagedInstances.json:766]: @TimLovellSmith 
it seems that we need to let PM or owners sync to fix it in the next API version. 
We are targeting to release this version as soon as possible to unblock downstream client development for Ignite conference.

File: specification/sql/resource-manager/Microsoft.Sql/preview/2021-05-01-preview/ServerTrustCertificates.json (hunk 12 above)
Comment (ericshape) [reply specification/sql/resource-manager/Microsoft.Sql/preview/2021-05-01-preview/ServerTrustCertificates.json]: done

File: specification/sql/resource-manager/Microsoft.Sql/preview/2021-05-01-preview/ServerTrustCertificates.json (hunk 11 above)
Comment (ericshape) [reply specification/sql/resource-manager/Microsoft.Sql/preview/2021-05-01-preview/ServerTrustCertificates.json]: done

File: specification/sql/resource-manager/Microsoft.Sql/preview/2021-05-01-preview/ManagedInstanceOperations.json (hunk 3 above)
Comment (ericshape) [reply specification/sql/resource-manager/Microsoft.Sql/preview/2021-05-01-preview/ManagedInstanceOperations.json:342]: added description in 6a3633278a825c2d19fc8edfc37650175638bb24

File: specification/sql/resource-manager/Microsoft.Sql/preview/2021-05-01-preview/ManagedInstances.json (hunk 4 above)
Comment (TimLovellSmith) [reply specification/sql/resource-manager/Microsoft.Sql/preview/2021-05-01-preview/ManagedInstances.json:766]: @ericshape I still don't understand why you would postpone this one. I think you have time to fix it; it should not be hard to update the swagger in the PR to fix.

Comment (ericshape) [reply specification/sql/resource-manager/Microsoft.Sql/preview/2021-05-01-preview/ManagedInstances.json]: addressed and update the enum name

**Review by zhenglaizhang — APPROVED** [review]

**Comment by zhenglaizhang** [comment]
.net sdk failure is expected
